src/components/views/Inbox.vue

```diff
@@ -32,22 +32,11 @@ export default {
     }
   },
   mounted() {
-    StudentQuestionService.getStudentQuestions(this, {})
-      .then(
-        (questions) => {
-          questions.map((q) => {
-            let o = {};
-              o.topic = q.topic;
-              o.subTopic = q.subTopic;
-              o.student = {};
-              o.student.name = q.student.name;
-              o.student.picture = q.student.picture;
-              o._id = q._id;
-
-            this.helpRequests.requests.push(o);
-          });
-        }
-      );
+    StudentQuestionService.getStudentQuestions(this, {}).then((questions) => {
+      questions.map((question) => {
```

Comment:
Instead of mapping and pushing each question, you can simply use object spread / concat:

`this.helpRequests.requests = [this.helpRequest.requests, ...questions]`

Reply:
It'd be ideal, but for some [weird JS limitation](https://vuejs.org/v2/guide/list.html#Caveats), it doesn't work because it lacks reactivity (i.e.: the UI won't update).

I also tried [using `Vue.set()`](https://www.reddit.com/r/vuejs/comments/4p9fre/forcing_refresh_of_vfor_when_updating_nested/) or even [adding a (hacky) empty `.push()`](https://stackoverflow.com/questions/42807888/vuejs-and-vue-set-update-array) after the direct assignment but none of this worked, so for now we'll have to stick with the verbose `.map()`.